### PR TITLE
Include virtual folder items in recursive resource/{id}/items.

### PR DESCRIPTION
**Pull request by manthey**
This also removes some duplicated code.

### CODE REVIEW

**Comment by manthey** [comment]
@sgratzl If you'd like to review this PR, that would be great.

File: server/__init__.py [hunk 1]
```diff
@@ -17,32 +17,37 @@
...
-    except ImportError:
-        return  # If virtual folders aren't enabled, do nothing
-    params['folderId'] = event.info['id']
-    return girder.plugins.virtual_folders._virtualChildItems(event)
+def childItems(self, folder, limit=0, offset=0, sort=None, filters=None,
+               includeVirtual=False, **kwargs):
+    if not includeVirtual or not folder.get('isVirtual') or 'virtualItemsQuery' not in folder:
+        return originalChildItems(self, folder, limit=limit, offset=offset,
+                                  sort=sort, filters=filters, **kwargs)
+    q = json_util.loads(folder['virtualItemsQuery'])
```
Comment (sgratzl) [inline server/__init__.py]: as far as I know the plugin handles cases in which is is an invalid JSON, like wrapping with a proper try/catch

File: server/__init__.py [hunk 2]
```diff
@@ -17,32 +17,37 @@
...
+    if 'virtualItemsSort' in folder and sort is None:
+        sort = json.loads(folder['virtualItemsSort'])
+    q.update(filters or {})
```
Comment (sgratzl) [inline server/__init__.py]: wouldn't it be better to combine them using a logical and?

as in
```python
if filters:
  q = {
    '$and': [q, filters ]
  }
```

Comment (manthey) [reply server/__init__.py]: This mimics the code that is in girder core (https://github.com/girder/girder/blob/master/girder/models/folder.py#L410).  At least there, the intent seems to be able to replace part of the query expression.  Perhaps girder core should change, but I'd rather have this be the same as what is there.

File: server/__init__.py (hunk 1 above)
Comment (manthey) [reply server/__init__.py]: The stored query is validated on creation (https://github.com/girder/girder/blob/c565081e65d589e896fca4f98c8b32e9794df051/plugins/virtual_folders/girder_virtual_folders/__init__.py#L40).  In the virtual folder plugin, this is assumed and not guarded in use.

File: server/__init__.py (hunk 2 above)
Comment (sgratzl) [reply server/__init__.py]: see https://github.com/girder/girder/issues/3130

Comment (sgratzl) [reply server/__init__.py]: since the issue has been closed, I will resolve this issue to be complaint with the rest of girder

**Pull request merged**